back/engines/commercial/id_vienna_saml/app/lib/id_vienna_saml/id_vienna_saml_omniauth.rb

```diff
@@ -0,0 +1,74 @@
+module IdViennaSaml
+  # Provides a SAML Omniauth configuration for Vienna's StandardPortal.
+  class IdViennaSamlOmniauth
+    FIXED_METADATA = { issuer: 'CitizenLab' }.freeze
+
+    # Extracts user attributes from the Omniauth response auth.
+    # @param [OmniAuth::AuthHash] auth
+    # @return [Hash] The user attributes
+    def profile_to_user_attrs(auth)
+      attrs = auth.dig(:extra, :raw_info).to_h
+
+      {
+        email: attrs.fetch('urn:oid:0.9.2342.19200300.100.1.3').first,
+        first_name: attrs.fetch('urn:oid:2.5.4.42').first,
+        last_name: attrs.fetch('urn:oid:1.2.40.0.10.2.1.1.261.20').first,
+        locale: AppConfiguration.instance.settings.dig('core', 'locales').first
```

Comment:
This will only be used when a new user record is created. The method `updatable_user_attrs` specifies which ones can be updated on subsequent logins (locale is not in there).